sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -263,6 +267,119 @@ case class GpuStringTrimRight(column: Expression, trimParameters: Option[Express
+      case sepScalar: GpuScalar =>
+        withResource(GpuScalar.from("", StringType)) { emptyStrScalar =>
+          GpuColumnVector.from(cv.stringConcatenateListElements(sepScalar.getBase, emptyStrScalar,
+            false, false), dataType)
+        }
+      case sepVec: GpuColumnVector =>
+        GpuColumnVector.from(cv.stringConcatenateListElements(sepVec.getBase), dataType)
+    }
+  }
+
+  private def processSingleColScalarSep(cv: ColumnVector): GpuColumnVector = {
+    // single column with scalar separator just replace any nulls with empty string
+    withResource(Scalar.fromString("")) { emptyStrScalar =>
+      GpuColumnVector.from(cv.replaceNulls(emptyStrScalar), dataType)
+    }
+  }
+
+  private def stringConcatSeparatorScalar(columns: ArrayBuffer[ColumnVector],
+      sep: GpuScalar): GpuColumnVector = {
+    withResource(Scalar.fromString("")) { emptyStrScalar =>
+      GpuColumnVector.from(ColumnVector.stringConcatenate(sep.getBase, emptyStrScalar,
+        columns.toArray[ColumnView], false), dataType)
+    }
+  }
+
+  private def stringConcatSeparatorVector(columns: ArrayBuffer[ColumnVector],
+      sep: GpuColumnVector): GpuColumnVector = {
+    // GpuOverrides doesn't allow only specifying a separator, you have to specify at
+    // least one value column
+    GpuColumnVector.from(ColumnVector.stringConcatenate(columns.toArray[ColumnView],
+      sep.getBase), dataType)
+  }
+
+  private def resolveColumnVectorAndConcatArrayCols(
+      expr: Expression,
+      numRows: Int,
+      colOrScalarSep: Any,
+      batch: ColumnarBatch): GpuColumnVector = {
+    withResourceIfAllowed(expr.columnarEval(batch)) {
+      case vector: GpuColumnVector =>
+        vector.dataType() match {
+          case ArrayType(st: StringType, nullable) => concatArrayCol(colOrScalarSep, vector.getBase)
+          case _ => vector.incRefCount()
+        }
+      case s: GpuScalar =>
+        s.dataType match {
+          case ArrayType(st: StringType, nullable) =>
+            // we have to first concatenate any array types
+            withResource(GpuColumnVector.from(s, numRows, s.dataType).getBase) { cv =>
+              concatArrayCol(colOrScalarSep, cv)
+            }
+          case _ => GpuColumnVector.from(s, numRows, s.dataType)
+        }
+      case other =>
+        throw new IllegalArgumentException(s"Cannot resolve a ColumnVector from the value:" +
+          s" $other. Please convert it to a GpuScalar or a GpuColumnVector before returning.")
+      }
+  }
+
+  private def checkScalarSeparatorNull(colOrScalarSep: Any,
+      numRows: Int): Option[GpuColumnVector] = {
+    colOrScalarSep match {
+      case sepScalar: GpuScalar =>
+        if (!sepScalar.getBase.isValid()) {
+          // if null scalar separator just return a column of all nulls
+          Some(GpuColumnVector.from(sepScalar, numRows, dataType))
+        } else {
+          None
+        }
+      case sepVec: GpuColumnVector =>
+        None
+    }
```

Comment:
nit: this could be simplified 
```suggestion
    colOrScalarSep match {
      case sepScalar: GpuScalar if !sepScalar.getBase.isValid =>
        // if null scalar separator just return a column of all nulls
        Some(GpuColumnVector.from(sepScalar, numRows, dataType))
      case _ =>
        None
    }
```